controller/thymis_controller/task/__init__.py

```diff
@@ -385,6 +406,70 @@ def copy_for_retry(self):
         return new_task
 
 
+class NixCommandTask(Task):
+    nix_process: NixProcess
+
+    def __init__(self, display_name, args, nix_command=None, env=None):
+        super().__init__(display_name)
+
+        self.nix_process = NixProcess(args, nix_command, env)
```

Comment:
NixProcess constructor is `__init__(self, args, nix_command=None, cwd=None, env=None)`, thus env is passed as cwd.

Also (after fixing locally), passing any envs seem to be break the command. Currently only affects deploy

Reply:
Not sure what you mean by "break command"

Reply:
my bad, missed that PATH is also required when passing other arbitrary env variables